### PR TITLE
patch: remove dimension validation

### PR DESCRIPTION
**Pull request by saminahbab**
### What

Removed the validation for single dimension filters, allowing them to pass through. 
Ammend tests to represent new validation rules 

### How to review

Check tests 

### Who can review

team b

### CODE REVIEW

File: api/contract_test.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ func TestCreateFiltersRequestValid(t *testing.T) {
 			r.Dimensions = r.Dimensions[:1]
 			Convey("When Valid() is called", func() {
 				err := r.Valid()
-				So(err, ShouldNotBeNil)
+				So(err, ShouldBeNil)
```
Comment (j-s-rawat) [inline api/contract_test.go:76]: This test no longer make much sense. Will it be better to update the test where a filter with 0 dimensions return an error, else we can remove the test

Comment (saminahbab) [reply api/contract_test.go]: I just removed it

File: datastore/mongodb/filter_dimensions.go [hunk 2]
```diff
@@ -342,7 +342,7 @@ func (c *Client) DeleteFilterDimension(ctx context.Context, filterID, dimensionN
 
 	// check the available filter dimensions.
 	// 1. A filter should always have a minimum of 2 dimensions
-	if len(filter.Dimensions) < 3 {
+	if len(filter.Dimensions) < 1 {
```
Comment (ChrysmOre) [inline datastore/mongodb/filter_dimensions.go:345]: Should this have not gone down to 2 rather than 1? Also comment will need updating

Comment (saminahbab) [reply datastore/mongodb/filter_dimensions.go]: done

**Review by j-s-rawat — APPROVED** [review]
LGTM

**Pull request closed without merging**